bookstore/publish.py

```diff
@@ -33,20 +38,41 @@ def initialize(self):
             raise web.HTTPError(400, "Must provide a path for publishing")
 
         model = self.get_json_body()
-        if model:
-            await self._publish(model, path.lstrip('/'))
-        else:
-            raise web.HTTPError(400, "Cannot publish an empty model")
+        self.validate_model(model)
+        resp = await self._publish(model['content'], path.lstrip('/'))
+        self.finish(resp)
 
-    async def _publish(self, model, path):
-        """Publish notebook model to the path"""
-        if model['type'] != 'notebook':
-            raise web.HTTPError(400, "bookstore only publishes notebooks")
-        content = model['content']
+    def validate_model(self, model):
+        """Checks that the model given to the API handler meets bookstore's expected structure for a notebook.
 
-        full_s3_path = s3_path(
-            self.bookstore_settings.s3_bucket, self.bookstore_settings.published_prefix, path
-        )
+        Pattern for surfacing nbformat validation errors originally written in
+        https://github.com/jupyter/notebook/blob/a44a367c219b60a19bee003877d32c3ff1ce2412/notebook/services/contents/manager.py#L353-L355
+
+        Raises
+        ------
+        tornado.web.HTTPError
+            Your model does not validate correctly
+        """
+        if not model:
+            raise web.HTTPError(400, "Bookstore cannot publish an empty model")
+        if model.get('type', "") != 'notebook':
+            raise web.HTTPError(400, "Bookstore only publishes notebooks")
+
+        content = model.get('content', {})
+        if content == {}:
+            raise web.HTTPError(400, "Bookstore cannot publish empty contents")
```

Comment:
Minor, but should be a 422 (https://www.restapitutorial.com/httpstatuscodes.html)

Reply:
I can make the change, but I would have thought that we were expecting 'content' on the request — I suppose the tricky thing is that this check both would trigger if content is not present on the request (400) and if it is present but is an empty dict (422)…

Is it worth making that distinction?